lib/shared/blocs/context_menu_options/context_menu_options.bloc.dart

```diff
@@ -132,20 +133,25 @@ class ContextMenuOptionsBloc extends ChangeNotifier with ManifestConsumer, Littl
     return totalPower / itemCount;
   }
 
-  double _getAchievableAverage(double currentAverage, Map<int, DestinyItemInfo> maxPowerEquipment) {
-    double? achievable;
-    int iterations = 0;
+  // Find the highest achievable power level without getting above power rewards. This is
+  // to help players decide whether to go for on power or above power rewards when leveling up.
+  // As of season 20:
+  //   - When below the powerful cap, powerful rewards will drop above power level.
+  //   - When at or above the powerful cap, powerfuls drop at power level and only pinnacle
+  //     rewards will drop above level.
+  double _getAchievableAverage(Map<int, DestinyItemInfo> maxPowerEquipment) {
     final equipmentPower = maxPowerEquipment //
         .values
-        .map<double>((e) => (e.instanceInfo?.primaryStat?.value ?? 0).toDouble());
-    while (achievable == null || achievable > currentAverage || iterations > 300) {
-      if (achievable != null) currentAverage = achievable;
-      final totalPower = equipmentPower.fold<double>(0, (t, c) => t + c.clamp(currentAverage, double.maxFinite));
-      final itemCount = maxPowerEquipment.length;
-      achievable = totalPower / itemCount;
-      iterations++;
-    }
-    return achievable;
+        .map((e) => (e.instanceInfo?.primaryStat?.value ?? 0));
+    int totalPower = equipmentPower.fold(0, (t, c) => t + c);
+    final itemCount = maxPowerEquipment.length;
+    int currentBase;
+    int iterations = 300; // This shouldn't be necessary
```

Comment:
I agree, I added it just as a safe measure to avoid an infinite loop if it kept running forever on some corner case that I wasn't expecting.